generator_test.go

```diff
@@ -137,3 +138,51 @@ func TestSign(t *testing.T) {
+	client_email := "test@testing.iam.gserviceaccount.com"
+	userid := "tester"
+
+	if _, err := GenerateCustomToken(userid, nil, client_email, private_key); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestCustomAuthTokenInValidKey(t *testing.T) {
+	private_key := "" //Key must be PEM encoded PKCS1 or PKCS8 private key
+	client_email := "test@testing.iam.gserviceaccount.com"
+	userid := "tester"
+
+	if _, err := GenerateCustomToken(userid, nil, client_email, private_key); err == nil {
+		t.Fatal(err)
+	}
+}
+
+func TestCustomAuthTokenDeveloperClaims(t *testing.T) {
+	private_key := "-----BEGIN PRIVATE KEY-----\nMIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDXJ2PtihileBbe\nmgc4tckfZNf4ZXHYevhOJs7zj9lBxD4+CBvw9WWFBhnIvLhq42DoxPFz8jYbWsh9\nj7dFp9y0TDUiIIsvTm/KNJ9mL2ttgFiOZEdS9k8N1I9dDkM2UNi7WtGP03Tm0Abz\nfrvSCJKro3lr8+tzCzfV0rtkhIaurbOy3f+iy1Bh2dQ+m6bkc4IKFHeqKPEp46mp\nagGgxergN0WIHfhcjFXkL+GspcuRw8C+AX84GZHmefyYycXY1R0DkSWMTm5q1MUs\ny5rbiCNtHW/elXoVbnlvSka3+GM9HRFXrevF+BmJHk1AziOBy49R5fgmw92/LwNQ\n3KzJl5rBAgMBAAECggEAWov0TRrKJpE0prgSA/bVTsYE8j/XOrD94P4KKIzIdRoe\ny8Jj5/OOpv4bSdTKlAlfVnbT2uT7imWZbOZXzhPIGYTc86DYkq8i8ulUkA+y2WGj\nY0Gmlq6cNIjZUONYu/ooTCj7etkWILE5y63xY4JCH8PrrNf5pe/5rp5CSRpRCB/C\nZI1Arjy7ZUPaGypAH58gakDDBWIiSCRsFX+0bt5u+J+FWebcurYQ+Y+qdpJ6cKbZ\nen0AWHQroxpWw7dWwsSriUYroKJMKLt5zM/wOJ5N4qKVwpuJX6qWB7C6JyyOZaJY\nEM5oOdMl94/Ml3gUpyfWRwV2qJkSFsvwd895hOkrhQKBgQD30WauA0SiVcHQowu/\ngbszKGaWwkoELFhPxUrtNZrQdQYBBoqft5nEA0mzY5DSoDgs4SDTMJ8HlA79XOKd\nlqCBeKWxS10zGOIxx+qWPpkulRnQek1088ZBGBxDu8V+SVfEqH+sQLEN4H9lYNe0\nRCTxrLVGcv2bezv3rpC2vKFmQwKBgQDeQegi6a73UUJV3yYGM8VBfmZjUCGb5KdH\nH8b1nc+Ty7C0z7nXU1zizN9ZZaIVMivfSt8H2W+3ikQ4FY8f7DAAQmw2JFrvKb4N\n/nnQXGIcRfO/RdM8ui88Oh7uJgZrkE9SbaWoVlhzZOv64aLqi/TZAN5UstIpNLfA\nS9s2bz/EqwKBgQCkQBELMrVR1v8PtpE5y9V0ccmVEI8YNwANVxlzIT1L/tQM5/YH\nKBxtMzStBkfdoj25WTl1YFt3HWXV/bNheY1GYt2HJglOraZ2EifkjvbeTgp/CCDA\nbDYxvLY3GoQqUJgwivGcDICNTweA/O/a1fOajrrTR7HZVJOJdRULWPismwKBgDs1\njZT3chAayrQ7rVKLqioHdVlRuJJiOJn/Ai7eqrTx15JjoFuXrrAQ6hNTuvkwk3V5\n6a6ao2Ne50uVmrpjXmpDR7aourzp/uKVf3gdlFl53TSAcoTECN9fkGvbH2y6Vhdc\ndHxC/G9JXIBKae9X95Nz4sbnmIs3qxgEXVLEElXfAoGAMgmwfSWOcdBahUvpiwOu\nJ0vJH5rmPYtBoFdd418Ynji2O0/8R9N5gDshff7ns6NBL/u1OS23fESWs4Lv09Hh\nS/ExVyxe2sa1RyDVbljLH5+aLkpWCfiNEFGcJ0kyIucPzjHJLCC6dTKh84fSTPRm\nWRRCbVUmvLnwcpIpPMhcnTc=\n-----END PRIVATE KEY-----\n"
+	client_email := "test@testing.iam.gserviceaccount.com"
+	userid := "tester"
+
+	developerClaims := jws.Claims{}
+
+	developerClaims.Set("premium_account", true)
+
+	if _, err := GenerateCustomToken(userid, &developerClaims, client_email, private_key); err != nil {
```

Comment:
I think it would be a good idea to capture the token that is generated and make sure that it has [all the parts a custom token needs](https://firebase.google.com/docs/auth/server/create-custom-tokens#create_custom_tokens_using_a_third-party_jwt_library).